### PR TITLE
LPS-118086 Crypto Hash Library intefaces without builders and configurations + skeleton impls, instructed by Brian

### PR DESCRIPTION
**Pull request by arthurchan35**
https://github.com/brianchandotcom/liferay-portal/pull/97488#issuecomment-756903225

### CODE REVIEW

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/flavor/CryptoHashFlavorImpl.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/flavor/CryptoHashFlavorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/context/CryptoHashGenerationContextImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/context/CryptoHashGenerationContextImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/response/CryptoHashGenerationResponseImpl.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/response/CryptoHashGenerationResponseImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/CryptoHashProcessorImpl.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/CryptoHashProcessorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/verification/context/CryptoHashVerificationContextImpl.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/verification/context/CryptoHashVerificationContextImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by csierra** [comment]
hey Arthur, 
per conversation with Brian we are going to go with https://github.com/liferay-appsec/liferay-portal/pull/215

**Pull request closed without merging**